docs/csharp/codedoc.md

```diff
@@ -12,16 +12,464 @@ ms.devlang: dotnet
+        /// </summary>
+        /// <param name="a">The stuff to do</param>
+        public void DoStuff(int a)
+        {
+
+        }
+    }
+
+<a name="typeparam"></a>
+## &lt;typeparam&gt;
+
+The &lt;typeparam&gt; tag should be used in the comment for a generic type or method declaration to describe a type parameter. 
+Add a tag for each type parameter of the generic type or method.
+
+    /// <summary>
+    /// This class does something
+    /// </summary>
+    public class SomeClass
+    {
+        /// <summary>
+        /// Does generic stuff
+        /// </summary>
+        /// <typeparam name="T">The generic stuff to do</param>
+        public void DoGenericStuff<T>()
+        {
+
+        }
+    }
+
+<a name="typeparamref"></a>
+## &lt;typeparamref&gt;
+
+The &lt;typeparamref&gt; tag gives you a way to indicate that a word in the code comments, 
+for example in a &lt;summary&gt; or &lt;remarks&gt; block refers to a type parameter.
+
+    /// <summary>
+    /// This class does something
+    /// </summary>
+    public class SomeClass
+    {
+        /// <summary>
+        /// Does generic stuff <typeparamref name="T"/>
+        /// </summary>
+        /// <typeparam name="T">The generic stuff to do</param>
+        public void DoGenericStuff<T>()
+        {
+
+        }
+    }
+
+<a name="include"></a>
+## &lt;include&gt;
+
+The &lt;include&gt; tag lets you refer to comments in a separate XML file that describe the types and members 
+in your source code as opposed to placing documentation comments directly in your source code file.
+
+    
+    /// <include file='docs.xml' path='Docs/Members[@name="someclass"]/SomeClass/*' />
+    public class SomeClass
+    {
+        /// <include file='docs.xml' path='Docs/Members[@name="someclass"]/DoStuff/*' />
+        public void DoStuff(int a)
+        {
+
+        }
+    }
+
+    /// <include file='docs.xml' path='Docs/Members[@name="someotherclass"]/SomeOtherClass/*' />
+    public class SomeOtherClass
+    {
+
+    }
+
+    /*
+        Contents of docs.xml
+        --------------------
+
+        <Docs>
+            <Members name="someclass">
+                <SomeClass>
+                    <summary>
+                    This class does something
+                    </summary>
+                </SomeClass>
+                <DoStuff>
+                    <summary>
+                    This method does stuff
+                    </summary>
+                </DoStuff>
+            </Members>
+            <Members name="someotherclass">
+                <SomeOtherClass>
+                    <summary>
+                    This class does some other thing
+                    </summary>
+                </SomeOtherClass>
+            </Members>
+        </Docs>
+    */
+
+The `filename` attribute represents the name of the XML file containing the documentation.
+
+The `tagpath` attribute represents the path of the tags in `filename` that leads to the `tag name`.
```

Comment:
Is it `path` or `tagpath`? The code uses one, here you're talking about the other.

Also, the path looks like XPath, maybe mention that and add some link to it, to explain what syntax can the path be?